newlib/libc/include/sys/config.h

```diff
@@ -266,6 +266,29 @@ SUCH DAMAGE.
 #define NEWLIB_THREAD_LOCAL
 #endif
 
+#ifdef _HAVE_HAS_BUILTIN
```

Comment:
Not sure this needs to be checked at configure time: https://godbolt.org/z/aYGaGWrfc
```suggestion
#if !defined(__has_builtin)
#define __has_builtin(...) 0
#endif
```

Reply:
Before GCC 10, gcc didn't have __has_builtin. So, I auto-detect the presence of `__has_builtin` in meson.build, when not present, it then goes and detects all of the builtins used by the library for that case. Otherwise, yeah, any other compiler without `__has_builtin` probably doesn't have builtins at all.

Reply:
Thanks for taking a look at this; I'm slowly approaching the ability to build an aarch64/arm multilib config with clang.

Reply:
If the compiler supports __has_builtin then it will be defined as a macro. My godbolt link shows this works as expected for GCC < 10

Reply:
Oh, right, the issue is that we still need to do all of the builtin detection at meson time, but we don't need another define to use at compile time. One moment.